WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewController.swift

```diff
@@ -105,15 +108,50 @@ private extension DashboardViewController {
         navigationItem.backBarButtonItem = backButton
     }
 
-    func configureDashboardUI() {
+    func configureDashboardUIContainer() {
         // A container view is added to respond to safe area insets from the view controller.
         // This is needed when the child view controller's view has to use a frame-based layout
         // (e.g. when the child view controller is a `ButtonBarPagerTabStripViewController` subclass).
         view.addSubview(containerView)
         containerView.translatesAutoresizingMaskIntoConstraints = false
         view.pinSubviewToSafeArea(containerView)
+    }
+
+    func reloadDashboardUIStatsVersion() {
+        guard let siteID = ServiceLocator.stores.sessionManager.defaultStoreID else {
+            return
+        }
+        if siteID != self.siteID {
+            dashboardUIFactory = DashboardUIFactory(siteID: siteID, stores: ServiceLocator.stores)
```

Comment:
I naturally went for DI with the constructor, assuming the stores manager has the same lifetime of the dashboard UI per site. Is there anything that could break this assumption? just thought we can start DI'ing from smalle places for the benefits of DI :) but there might be things I don't know about `StoresManager` that could break the assumption

Reply:
To me, it's not about the cases when that assumption would break, bu the fact that we are assuming a given lifecycle. The StoresManager has always been a "fire and forget" API, and I am reluctant to change that.

In terms of DI itself, I would not inject any of the dependencies provided by the ServiceLocator, and would definitely inject all others. In this case, for example, I would definitely inject the siteID, but I would rely on the ServiceLocator to get the reference to the stores manager. Same with StatsVersionStateCoordinator.

I guess my point, aside from the lifecycle, is that we have a way to fetch dependencies that provides abstractions, not concretions, which allows us to mock depdenecies, and is the spirit of the dependency inversion principle 😊

Reply:
Sorry, that was too long and a bit too abstract. Let me clarify, please... The way I see it, injecting, for example, a viewmodel into a view controller via the constructor is 💯 , however, I think we should avoid injecting also the stores manager, the analytics, the notices manager and other potential global dependencies when we have a way to pull abstractions representing them

Reply:
I see, it makes sense now given the "fire and forget" nature of API usage. I was wondering about the pros and cons of such API vs. a dependency object of an expected lifecycle (like an object that is always alive while the user is logged in to a given site)? I felt it's nice to use an object instead of a static global instance (or the global part abstracted) so that the object can have clear and flexible ownerships. The third-party libraries are usually global singletons probably for ease of use for all kinds of projects. We can move it to a separate discussion!

updated the stores manager to use `ServiceLocator`'s in https://github.com/woocommerce/woocommerce-ios/pull/1239/commits/a3ec909ad3fa6be3c62ee7ce418bd31c4f7edf9f